### PR TITLE
Tgp 2867/update record endpoint

### PR DESCRIPTION
**Pull request by kenneth-rayner**
# Checklist for developers before reviewing
 - [x] I have merged in main
 - [x] I have assigned the PR to me
 - [x] I have run `precheck.sh` to run all tests, format and check the code coverage
 - [x] I have written unit test and integration tests to cover any code that I wrote, where necessary
 - [x] I have linked tickets to any TODOs I have written
 - [x] I have run the microservices with the latest SM2 config
 - [x] I have checked all the Acceptance Criteria pass
 - [x] I have run the local trader-goods-profile-api-tests
 - [x] I have spoken to the QAs about any changes that may need to be made to the api tests OR no changes to the api tests are necessary
 - [x] I have moved the ticket to Review in Jira
 - [x] I have left a message in tgp-prs on slack to alert other Devs that this ticket is ready for review
 - [x] I have checked my PR is not in draft mode

# Checklist for reviewer 1
- [ ] I have added an emoji to the relevant message in tgp-prs to show I am reviewing
- [ ] I have checked out the branch
- [ ] I have run the microservices with the latest SM2 config
- [ ] I have checked all the Acceptance Criteria pass
- [ ] I reviewed the code and left comments where necessary
- [ ] I have left a message on the relevant tgp-prs message to show that I have finished reviewing

# Checklist for reviewer 2
- [ ] I have added an emoji to the relevant message in tgp-prs to show I am reviewing
- [ ] I have checked out the branch
- [ ] I have run the microservices with the latest SM2 config
- [ ] I have checked all the Acceptance Criteria pass
- [ ] I reviewed the code and left comments where necessary
- [ ] I have left a message on the relevant tgp-prs message to show that I have finished reviewing

# Checklist for developers before merging
 - [ ] The PR has been approved by 2 other developers
 - [ ] I have merged in main
 - [ ] I have run `precheck.sh` to run all tests, check formatting and code coverage
 - [ ] I have run the microservices with the latest SM2 config
 - [ ] I have checked all the Acceptance Criteria pass
 - [ ] I have checked that the QAs have merged in any changes to the api tests
 - [ ] I have run the local trader-goods-profile-api-tests

# Reminders for developers after merging
 1. Wait for pipeline to pass
 2. Move ticket to QA column in Jira
 3. Leave message in tgp-qa on slack to alert QAs that this ticket is ready for testing

### CODE REVIEW

File: app/uk/gov/hmrc/tradergoodsprofilesrouter/config/HawkInstanceConfig.scala [hunk 1]
```diff
@@ -72,11 +76,13 @@ object HawkInstanceConfig {
         config.get[String]("create-record"),
         config.get[String]("remove-record"),
         config.get[String]("update-record"),
```
Comment (besscerule) [inline app/uk/gov/hmrc/tradergoodsprofilesrouter/config/HawkInstanceConfig.scala:78]: do we need this line anymore?

File: conf/application.conf [hunk 2]
```diff
@@ -98,15 +98,16 @@ microservice {
         host = localhost
         port = 10903
         forwarded-host = "MDTP"
-
         get-records  = "/tgp/getrecords/v1"
         create-record  = "/tgp/createrecord/v1"
         remove-record  = "/tgp/removerecord/v1"
         update-record  = "/tgp/updaterecord/v1"
```
Comment (besscerule) [inline conf/application.conf:104]: do we need this?

File: app/uk/gov/hmrc/tradergoodsprofilesrouter/config/AppConfig.scala [hunk 3]
```diff
@@ -38,10 +38,5 @@ class AppConfig @Inject() (config: Configuration) {
 
   lazy val sendAcceptHeader: Boolean = config.getOptional[Boolean]("features.sendAcceptHeader").getOrElse(true)
 
-  lazy val useEisPatchMethod: Boolean =
```
Comment (taylors-version) [inline app/uk/gov/hmrc/tradergoodsprofilesrouter/config/AppConfig.scala:41]: This needs pulling back until EIS Patch is in Prod

File: app/uk/gov/hmrc/tradergoodsprofilesrouter/connectors/UpdateRecordConnector.scala [hunk 4]
```diff
@@ -43,53 +43,23 @@ class UpdateRecordConnector @Inject() (
     correlationId: String
   )(implicit hc: HeaderCarrier): Future[Either[EisHttpErrorResponse, CreateOrUpdateRecordEisResponse]] = {
     val url = appConfig.hawkConfig.updateRecordUrl
-
-    //Todo: remove this flag when EIS has implemented the PATCH method - TGP-2417.
-    if (appConfig.useEisPatchMethod) {
```
Comment (taylors-version) [inline app/uk/gov/hmrc/tradergoodsprofilesrouter/connectors/UpdateRecordConnector.scala:48]: As mentioned, we still need this config for now

File: conf/application.conf [hunk 5]
```diff
@@ -136,7 +137,6 @@ microservice {
 
 features {
      niphlPaddingEnabled = false
-     useEisPatchMethod = false
```
Comment (taylors-version) [inline conf/application.conf:139]: Still need this too

**Review by taylors-version — CHANGES_REQUESTED** [review]
Looks good other than still needing the feature flag until everything goes to prod

File: app/uk/gov/hmrc/tradergoodsprofilesrouter/connectors/UpdateRecordConnector.scala [hunk 6]
```diff
@@ -68,7 +68,7 @@ class UpdateRecordConnector @Inject() (
     payload: UpdateRecordPayload,
```
[the remark this answers is not in the document]
Comment (kenneth-rayner) [reply app/uk/gov/hmrc/tradergoodsprofilesrouter/connectors/UpdateRecordConnector.scala:68]: We thought that this had to change to have the new split out bearer token. Have pushed up a small refactor.

[the remark this answers is not in the document]
Comment (taylors-version) [reply app/uk/gov/hmrc/tradergoodsprofilesrouter/connectors/UpdateRecordConnector.scala]: I've pushed the change that I think is needed. Let's have a chat tomorrow if you think I'm wrong

[the remark this answers is not in the document]
Comment (Reece-Carruthers) [reply app/uk/gov/hmrc/tradergoodsprofilesrouter/connectors/UpdateRecordConnector.scala]: Okay, just one thing I want to double check then is if we are sending the correct bearer tokens around, we made a new one for patch, but just want to be sure that the old one should be used for any connector call doing .put? and any doing .patch needs to use the new one?

[the remark this answers is not in the document]
Comment (taylors-version) [reply app/uk/gov/hmrc/tradergoodsprofilesrouter/connectors/UpdateRecordConnector.scala]: I have to check this one every single time. Here's the response I got from EIS:
> EIS will provide a new bearer token for new PUT request (/tgp/puttgprecord/v1). The existing bearer token should work for new PATCH request  (/tgp/update/record/v1). Hope it helps.
So the bearer token inside _our_ .patch remains the same (appConfig.hawkConfig.updateRecordUrl) and _our_ .put will use a new one.
So yea, I think the buildHeaderWithoutClientId needs to change

**Review by taylors-version — APPROVED** [review]

**Review by taylors-version — APPROVED** [review]

**Review by Ewan-James-Donovan — APPROVED** [review]

**Review by yohan-oc — APPROVED** [review]

**Pull request merged**